src/batch-client/index.ts

```diff
@@ -254,6 +254,28 @@ export class ZimbraBatchClient {
 			body: options
 		});
 
+	public downloadMessage = ({ id }: any) => {
+		return fetch(`${this.origin}/service/home/~/?auth=co&id=${id}`, {
+			method: 'GET',
+			headers: {
+				'X-Zimbra-Encoding': 'x-base64'
+			},
+			credentials: 'include'
+		}).then(response => {
+			if (response.ok) {
+				return response.text().then(content => {
+					if (!content) {
+						return undefined;
+					}
+					return {
+						id,
+						content
+					};
+				});
+			}
```

Comment:
Can this `return Promise.reject` with some information instead of with an implicit `return Promise.resolve(undefined)`?

Reply:
since it is returning then of .text(), it doesn't allow return Promise.reject.